### PR TITLE
feat: prefer `XDG_CONFIG_HOME` for `revive.toml` config file

### LINKED ISSUE #724: prefer XDG_CONFIG_HOME for getting the revive.toml config file
**Is your feature request related to a problem? Please describe.**
A lot of linux programs prefer `XDG_CONFIG_HOME` for default configuration. So a lot of linux users might prefer the default `XDG_CONFIG_HOME` instead of `HOME`.

**Describe the solution you'd like**
Check for the `revive.toml` file in `XDG_CONFIG_HOME` first and then check for it in `HOME`(for backward compatibility)

### PR DESCRIPTION
**Pull request by notjedi**
motivation: #724 

### Checklist
- [ ] Did you add tests?
- [x] Does your code follows the coding style of the rest of the repository?
- [x] Does the Travis build passes?

Closes #724

### CODE REVIEW

**Comment by notjedi** [comment]
i have only tested this on my `linux` machine, should most prolly work on `darwin` and `plan9`.

File: cli/main.go [hunk 1]
```diff
@@ -104,12 +104,22 @@ Example:
...
+	}
+
+	if _, err := os.Stat(configDirFile); err == nil {
+		result = configDirFile
+	} else if _, err := os.Stat(homeDirFile); err == nil {
+		result = homeDirFile
+	} else {
+		result = ""
 	}
+    print(result)
```
Comment (mgechev) [inline cli/main.go:122]: This here seems unnecessary.

**Review by mgechev — CHANGES_REQUESTED** [review]
We can remove the log statement and merge.

File: cli/main.go (hunk 1 above)
Comment (notjedi) [reply cli/main.go:122]: sorry my bad, left that there by mistake.

**Comment by notjedi** [comment]
i probably forgot about this pr and deleted the repo. my bad, can we pick up where we left? @mgechev

**Comment by mfederowicz** [comment]
@notjedi @mgechev this PR can be closed because all features related this topic we merged in #912 today :)

**Comment by chavacava** [comment]
Duplicated of #912

**Pull request closed without merging**